pilot/pkg/model/authentication.go

```diff
@@ -70,7 +73,37 @@ func GetConsolidateAuthenticationPolicy(store IstioConfigStore, service *Service
 	return nil
 }
 
-// ConstructSdsSecretConfig constructs SDS Sececret Configuration.
+// ConstructSdsSecretConfig constructs SDS secret configuration for ingress gateway.
```

Comment:
Can you reuse the other method that constructs the config for regular SDS ?

Is the expectation that Ingress will connect to 2 SDS servers ? How is the config different ?

Reply:
sdsconfig case for ingress agent case is much easier, using separated function will reduce the risk of causing the regression on existing sidecar case, and code is much cleaner.

Reply:
Yes, Ingress gateway connects to 2 SDS servers. As discussed above, specifying two separate paths is safer. The gRPC connections to those servers are different, the connection to Citadel agent requires channel credential and call credential (tokens) and the connection to ingress gateway agent does not require those credentials.